### PR TITLE
HardwareSerial refactoring

### PR DESCRIPTION
**Pull request by slaff**
The old code was causing constant stopping and starting of interrupts leading to bad performance.
It was not possible to use UART1 for debug output.
it was not possible to change / specify UART configuration or rx buffer size.

The new code, "borrowed" from esp8266/Arduino and adapted for Sming  addresses those issues. In addition it allows to address up to 256 uarts and attach callbacks for received data.

### CODE REVIEW

**Comment by slaff** [comment]
@ADiea Can you check this PR if it is working for you? It would be great if on top of that PR we can add your commandExecutor refactoring.

File: Sming/system/include/espinc/uart.h [hunk 1]
```diff
@@ -1,96 +1,168 @@
...
+#define UART_5O1 ( UART_NB_BIT_5 | UART_PARITY_ODD  | UART_NB_STOP_BIT_1 )
+#define UART_6O1 ( UART_NB_BIT_6 | UART_PARITY_ODD  | UART_NB_STOP_BIT_1 )
+#define UART_7O1 ( UART_NB_BIT_7 | UART_PARITY_ODD  | UART_NB_STOP_BIT_1 )
+#define UART_8O1 ( UART_NB_BIT_8 | UART_PARITY_ODD  | UART_NB_STOP_BIT_1 )
+#define UART_5O2 ( UART_NB_BIT_5 | UART_PARITY_ODD  | UART_NB_STOP_BIT_2 )
+#define UART_6O2 ( UART_NB_BIT_6 | UART_PARITY_ODD  | UART_NB_STOP_BIT_2 )
+#define UART_7O2 ( UART_NB_BIT_7 | UART_PARITY_ODD  | UART_NB_STOP_BIT_2 )
+#define UART_8O2 ( UART_NB_BIT_8 | UART_PARITY_ODD  | UART_NB_STOP_BIT_2 )
+
+/*
```
Comment (ADiea) [inline Sming/system/include/espinc/uart.h]: pls remove

**Comment by ADiea** [comment]
@slaff actually there is no extra work done by me on commandexecutor, just guarded it with #ifdefs to remove it from several places because together with String implementation was taking irom space - so just a DISABLE_COMMANDEXEC makefile switch...

**Comment by ADiea** [comment]
Could you modify basic serial sample to make use of the 2 uarts ex to output debug on uart1tx and setup a callback for uart0rx + echo on uart0tx. This way I could quickly test with 2x USB ft232 and confirm

**Comment by slaff** [comment]
@ADiea Check the latest version. In this diff there is more information and ready to use code (https://github.com/SmingHub/Sming/pull/894/commits/2d10332f25acd72d9d5c42b14b76141c3a63e64f).

**Comment by ADiea** [comment]
Will try to test this evening looks good:D

**Pull request merged**